lib/puppet/provider/package/yum.rb

```diff
@@ -159,6 +172,11 @@ def install
         self.debug "Downgrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
         operation = :downgrade
       end
+
+      if is && rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure])) > 0
```

Comment:
Not quite, though the outer `if` statement does clean things up. I was thinking of this though:

```
if is
        if rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure])) < 0
          self.debug "Downgrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
          operation = :downgrade
        elsif rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure])) > 0
          self.debug "Upgrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
          operation = update_command
        end
      end
```

Those two different conditionals can't both be true, but a casual reading of the code with the independent conditionals doesn't indicate this.